Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -24,6 +30,132 @@ namespace AzToolsFramework
     // temporary value until there is UI to expose the fields
     static constexpr int DefaultViewBookmarkCount = 12;
 
+    struct ViewBookmarkVisitor : AZ::SettingsRegistryInterface::Visitor
+    {
+        ViewBookmarkVisitor()
+            : m_viewBookmarksKey{ ViewBookmarksRegistryPath } {};
+
+        AZ::SettingsRegistryInterface::VisitResponse Traverse(
+            AZStd::string_view path,
+            [[maybe_unused]] AZStd::string_view valueName,
+            AZ::SettingsRegistryInterface::VisitAction action,
+            [[maybe_unused]] AZ::SettingsRegistryInterface::Type type) override
+        {
+            if (action == AZ::SettingsRegistryInterface::VisitAction::Begin)
+            {
+                // strip off the last json pointer key from the path and if it matches the view bookmark key then add an entry
+                // to the view bookmark map
+                AZStd::optional<AZStd::string_view> localBookmarksId = AZ::StringFunc::TokenizeLast(path, "/");
+                if (path == m_viewBookmarksKey && localBookmarksId.has_value() && !localBookmarksId->empty())
+                {
+                    if (const auto existingBookmarkEntry = m_bookmarkMap.find(localBookmarksId.value());
+                        existingBookmarkEntry == m_bookmarkMap.end())
+                    {
+                        m_bookmarkMap.insert(AZStd::make_pair(localBookmarksId.value(), AZStd::vector<ViewBookmark>{}));
+                    }
+                }
+            }
+
+            return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+        }
+
+        using AZ::SettingsRegistryInterface::Visitor::Visit;
+        void Visit(AZStd::string_view path, AZStd::string_view valueIndex, AZ::SettingsRegistryInterface::Type, double value) override
+        {
+            AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> dataType = AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> bookmarkIndexStr = AZ::StringFunc::TokenizeLast(path, "/");
+            AZStd::optional<AZStd::string_view> bookmarkType;
+            if (bookmarkIndexStr == LastKnownLocationKey)
```

Comment:
`bookmarkIndexStr` is an optional, shouldn't the string comparison be `if (*bookmarkIndexStr == LastKnownLocationKey)`

Reply:
I think this is actually safe as the `==` operator will check the underlying type (again I'll check tomorrow but I don't think this is actually wrong)

Reply:
I double checked and this is fine, no temporary is created as std::optional provides an overload for the equality operator which takes whatever type is inside the optional (see all the overloads here - https://en.cppreference.com/w/cpp/utility/optional/operator_cmp). So in this example...

```c++
std::string hello = "hello";
std::optional<std::string> optional_hello = "hello maybe";

if (hello == optional_hello) {
    // do something
}
```

The call is just `std::operator==(hello, optional_hello)`

See CppInsights for a [nice example](https://cppinsights.io/s/4dbc93b0)

Reply:
Thanks for checking. It's good to know :)